src/__tests__/vuetify.js

```diff
@@ -69,5 +69,5 @@ test('opens a menu', async () => {
   expect(menuItem).toBeInTheDocument()
 
   await fireEvent.click(menuItem)
-  expect(queryByText('menu item')).not.toBeInTheDocument()
+  expect(queryByText('menu item')).not.toBeVisible()
```

Comment:
Hm, did this test start to fail? 🤔  I mean, did Vuetify changed anything? I'm worried this is related to `attachTo`

Reply:
Yeah it did start fail. When I debugged though it seemed like once menu list was open it just added display none to hide it.
Let me debug with `attachToDocument` to see what it was outputting

Reply:
So with `master` this works fine. Using debug shows that the `menu item` is removed from the DOM.
However, with the latest version of VTU and `attachToDocument` still show this in the DOM.

So looks like an issue with the latest release of VTU rather than use of `attachTo`

Reply:
Ok, I spent some time on this. The breaking change is actually from  `vue/test-utils@1.0.0-beta.32` -> `vue/test-utils@1.0.0-beta.33`.

I will debug a bit more.

Reply:
Don't ask how I figured this out but [THIS LINE](https://github.com/vuejs/vue-test-utils/compare/v1.0.0-beta.32...v1.0.0-beta.33?diff=unified#diff-7881cb3665479e4aa8177337f2ccfbc9L24) is the problem. Adding it back in fixes this test.

I do not know why this was removed or what this line does.

Reply:
Looks like the line was removed in https://github.com/vuejs/vue-test-utils/pull/1479. Looks like it fixes an extreme (sic) edge-case, but I have a hard time finding the relationship with this test 😅 I'm a bit worried, though, as the test _looked fine_ and I believe we're following the suggested approach to test Vuetify components. Weird, though, that noone reported similar issues 🤔

Reply:
Vuetify is quite unique in how it extends many components to make others - something along those lines, perhaps.

VTU v1 drives my crazy for this reason. Why so many hacks? Just mount the damn component, lol.

I think someone (I can look into it, but happy if someone else wants to help) needs to dive into that snippet and figure out what it actually does that changes this.

Reply:
@lmiller1990 @afontcu I'm happy to dive into the specific component in Vuetify to see whether any weirdness is occurring that may throw off VTU.
Have either of you had any more thoughts on whether the cause in VTU could be modified?

Reply:
Hey! 👋 yeah, I'm currently looking into it. I started to think that `toBeInTheDocument()` was actually not the right assertion.

AFAICT, Vuetify does render the element but hides it, so `toBeInTheDocument()` should have passed. Now, I don't have much experience with Vuetify, so I might be wrong.

I am unable, though, to make this pass (and I think it should):

```js
test('opens a menu', async () => {
  const {getByText, queryByText} = renderWithVuetify(VuetifyDemoComponent)

  const openMenuButton = getByText('open menu')

  // Menu item is initially not rendered
  expect(queryByText('menu item')).not.toBeInTheDocument()

  await fireEvent.click(openMenuButton)

  const menuItem = getByText('menu item')

  expect(menuItem).toBeInTheDocument()
  expect(menuItem).toBeVisible() // <-- this fails (it still has "display: none;")

  await fireEvent.click(openMenuButton)

  expect(menuItem).not.toBeVisible()
})
```

Reply:
@afontcu so just tried the menu in a project I have locally, the browser behaviour is:

1. Mounted `v-menu` does not appear in the document
1. Users clicks activator. In my case a `v-text-field`. `v-menu` is inserted into the DOM
1. User clicks activator again to close the menu. `v-menu` is still in the DOM but has `display: none`

So the edit to the test looks valid based on this behaviour.
Looks like the test before was reporting a false positive.

Reply:
> Looks like the test before was reporting a false positive.

Yes! That's what I thought, too 🙌 . The snippet I posted above is currently failing, I'll try to dig a bit more and see if I can figure out why. Thought it could have something to with transitions, even though I tried with `<v-menu :transition="false" />` but nothing changed.

Reply:
Tried a lot of different versions of this test, then figured:

> "I wonder how Vuetify are testing the menu component"

Turns out the definition of this "working" is that it renders with `display: none` even when clicked whilst testing with VTU.

There is a snapshot here with `display: none` https://github.com/vuetifyjs/vuetify/blob/9ac1158f857fd1f531f1177ba623602973f81b55/packages/vuetify/src/components/VMenu/__tests__/__snapshots__/VMenu.spec.ts.snap#L15
With the corresponding test here https://github.com/vuetifyjs/vuetify/blob/9ac1158f857fd1f531f1177ba623602973f81b55/packages/vuetify/src/components/VMenu/__tests__/VMenu.spec.ts#L32

Reply:
okay, this is interesting. Now I'm leaning towards simplifying our Vuetify test (which is only aimed at showing a way of configuring VTL and Vuetify) to make it more resilient. If they are happy with `display:none`, we should be, too. I'm gonna push some changes soon to this branch, and make the test pass.

Thanks for your help, @afenton90! Much appreciated :)